x-pack/legacy/plugins/siem/public/pages/case/components/user_action_tree/index.test.tsx

```diff
@@ -0,0 +1,281 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+
+import { Router, routeData, mockHistory, mockLocation } from '../__mock__/router';
+import { getFormMock } from '../__mock__/form';
+import { useUpdateComment } from '../../../../containers/case/use_update_comment';
+import { basicCase, getUserAction } from '../__mock__/case_data';
+import { UserActionTree } from './';
+import { TestProviders } from '../../../../mock';
+import { useFormMock } from '../create/index.test';
+import { wait } from '../../../../lib/helpers';
+import { act } from 'react-dom/test-utils';
+jest.mock(
+  '../../../../../../../../../src/plugins/es_ui_shared/static/forms/hook_form_lib/hooks/use_form'
+);
+
+const fetchUserActions = jest.fn();
+const onUpdateField = jest.fn();
+const updateCase = jest.fn();
+const defaultProps = {
+  data: basicCase,
+  caseUserActions: [],
+  firstIndexPushToService: -1,
+  isLoadingDescription: false,
+  isLoadingUserActions: false,
+  lastIndexPushToService: -1,
+  userCanCrud: true,
+  fetchUserActions,
+  onUpdateField,
+  updateCase,
+};
+const useUpdateCommentMock = useUpdateComment as jest.Mock;
+jest.mock('../../../../containers/case/use_update_comment');
+
+const patchComment = jest.fn();
+describe('UserActionTree ', () => {
```

Comment:
Initial render test :)

Reply:
I think we should add the following tests:

- The description is being shown and can be edited.
- The correct number and type of `caseUserActions` is being shown.
- Props such as `isEditable`, `isLoading`, `fullname`, and `username` should be checked for correctness.
- Test if the `showTopFooter` and `showBottomFooter` renders correctly.

Reply:
- description render test happens in `case_view/index.test.tsx` and edit is tested in this file in `'calls update description when description markdown is saved'`
- I don't think this will improve the tests. We are already checking every scenario in the render for caseUserActions, every possible is covered in a test already
- some of these are covered
  - `isEditable` is tested in 'Switches to markdown when edit is clicked and back to panel when canceled'
  - `isLoading` is tested in `case_view/index.test.tsx` on line 245:
   ```
    expect(
      wrapper
        .find('[data-test-subj="description-action"] [data-test-subj="user-action-title-loading"]')
        .first()
        .exists()
    ).toBeTruthy();
- I'll add tests for these
  - `fullName`, `username`, `showTopFooter` and `showBottomFooter`